server/task/task-distributed/src/test/java/org/apache/james/task/eventsourcing/distributed/DistributedTaskManagerTest.java

```diff
@@ -299,4 +301,42 @@ void givenTwoTaskManagerAndATaskRanPerTaskManagerListingThemOnEachShouldShowBoth
                 });
         }
     }
+
+    @Test
+    void givenTwoTaskManagerIfTheFirstOneIsDownTheSecondOneShouldBeAbleToRunTheRemainingTasks(CountDownLatch countDownLatch) throws InterruptedException {
+        try (EventSourcingTaskManager taskManager1 = taskManager()) {
+            try (EventSourcingTaskManager taskManager2 = taskManager(HOSTNAME_2)) {
+                ImmutableBiMap<TaskManager, Hostname> hostNameByTaskManager = ImmutableBiMap.of(taskManager1, HOSTNAME, taskManager2, HOSTNAME_2);
+                TaskId firstTask = taskManager1.submit(new MemoryReferenceTask(() -> {
+                    countDownLatch.await();
+                    return Task.Result.COMPLETED;
+                }));
+
+                awaitUntilTaskHasStatus(firstTask, TaskManager.Status.IN_PROGRESS, taskManager1);
+
+                Hostname nodeRunningFirstTask = taskManager1.getExecutionDetails(firstTask).getRanNode().get();
+                Hostname otherNode = getOtherNode(hostNameByTaskManager, nodeRunningFirstTask);
+                EventSourcingTaskManager taskManagerRunningFirstTask = (EventSourcingTaskManager) hostNameByTaskManager.inverse().get(nodeRunningFirstTask);
+                EventSourcingTaskManager otherTaskManager = (EventSourcingTaskManager) hostNameByTaskManager.inverse().get(otherNode);
+
+                TaskId taskToExecuteAfterFirstNodeIsDown = taskManagerRunningFirstTask.submit(new CompletedTask());
+                taskManagerRunningFirstTask.close();
```

Comment:
why it's closed after the submit?

Reply:
To close the connection between, the current single active consumer and rabbitmq.
What is worrying you. You would prefer a sleep after the submit to ensure that submit is effectively done ? or submit the task from the other task manager ?

Reply:
maybe I don't understand but how do you ensure the taskManagerRunningFirstTask won't run this task as it's not yet closed when you submit it?

Reply:
Before i have not released the countdown blocking the execution of the first task.

Reply:
ok